data/actions/lib/actions.lua

```diff
@@ -172,7 +172,7 @@ function onUseShovel(player, item, fromPosition, target, toPosition, isHotkey)
 		tile:relocateTo(toPosition)
 	elseif table.contains(sandIds, groundId) then
 		local randomValue = math.random(1, 100)
-		if target.actionid == 100 and randomValue <= 20 then
+		if target.actionid == aId.sandHole and randomValue <= 20 then
 			ground:transform(489)
 			ground:decay()
 		elseif randomValue == 1 then
```

Comment:
if [this](https://github.com/otland/forgottenserver/pull/2968#issuecomment-619383257) is right, we should check for another action id here and on the last else.